src/daft-logical-plan/src/optimization/rules/push_down_filter.rs

```diff
@@ -87,6 +87,10 @@ impl PushDownFilter {
 
                     // Pushdown filter into the Source node
                     SourceInfo::Physical(external_info) => {
+                        // If the scan is materialized, we don't pushdown the filter.
+                        if matches!(external_info.scan_state, ScanState::Tasks(_)) {
+                            return Ok(Transformed::no(plan));
+                        }
```

Comment:
What is this for (as it relates to the fix)?

Reply:
Ah when I was looking at the query plans I realized I didn't do the projection and filter pushdowns after join ordering. Added those, then realized we can't/shouldn't do filter pushdowns into scan nodes that have already been materialized. We actually panic without this check because we later call `get_scan_op()` which is implemented as
```
    pub fn get_scan_op(&self) -> &ScanOperatorRef {
        match self {
            Self::Operator(scan_op) => scan_op,
            Self::Tasks(_) => panic!("Tried to get scan op from materialized physical scan info"),
        }
    }
```